fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryUnionPlan.java

```diff
@@ -176,8 +176,9 @@ public int hashCodeWithoutChildren() {
     }
 
     @Override
-    public int planHash() {
-        return PlanHashable.planHash(getQueryPlanChildren()) + getComparisonKey().planHash() + (isReverse() ? 1 : 0);
+    public int planHash(PlanHashKind hashKind) {
+        // TODO: Super()?
+        return PlanHashable.planHash(hashKind, getQueryPlanChildren()) + getComparisonKey().planHash(hashKind) + (isReverse() ? 1 : 0);
```

Comment:
It totally could call super me thinks! I you change it make sure you don't hash the children twice.

Reply:
Done.